tests/ContextIntegrationTests.js

```diff
@@ -0,0 +1,318 @@
+                    event: "{logoutRequest}.events.onComplete",
+                    listener: "gpii.test.logoutRequestListen"
+                }
+            ],
+            gpii.test.createProcessChecks(gpii.test.contextIntegration.data.processes, "expectRestored"),
+            [
+                {
+                    func: "gpii.test.checkRestoredConfiguration",
+                    args: ["{tests}.gpii-default.settingsHandlers", "{tests}.settingsStore", "{nameResolver}"]
+                }
+            ]
+        ]
+    }, {
+        name: "Multiple context changes",
+        expected: 8,
+        sequenceSegments: [
+            [
+                {
+                    "func": "gpii.test.expandSettings",
+                    args: [ "{tests}", [ "bright", "gpii-default" ]]
+                }, {
+                    func: "gpii.test.snapshotSettings",
+                    args: ["{tests}.gpii-default.settingsHandlers", "{tests}.settingsStore", "{nameResolver}"]
+                }, {
+                    func: "{loginRequest}.send"
+                }, {
+                    event: "{loginRequest}.events.onComplete",
+                    listener: "gpii.test.loginRequestListen"
+                }, {
+                    func: "gpii.test.checkConfiguration",
+                    args: ["{tests}.gpii-default.settingsHandlers", "{nameResolver}"]
+                }
+            ],
+            gpii.test.createProcessChecks(gpii.test.contextIntegration.data.processes, "expectConfigured"),
+            gpii.test.contextIntegration.changeEnvironmentAndCheck("bright"),
+            gpii.test.contextIntegration.changeEnvironmentAndCheck("noise"),
+            gpii.test.contextIntegration.changeEnvironmentAndCheck("brightandnoise"),
+
+            [
+                {
+                    func: "{logoutRequest}.send"
+                }, {
+                    event: "{logoutRequest}.events.onComplete",
+                    listener: "gpii.test.logoutRequestListen"
+                }
+            ],
+            gpii.test.createProcessChecks(gpii.test.contextIntegration.data.processes, "expectRestored"),
+            [
+                {
+                    func: "gpii.test.checkRestoredConfiguration",
+                    args: ["{tests}.gpii-default.settingsHandlers", "{tests}.settingsStore", "{nameResolver}"]
+                }
+            ]
+        ]
+    }
+];
+
+gpii.test.contextIntegration.buildTestFixtures = function (fixtures) {
+    return fluid.transform(fixtures, function (fixture) {
+        var testDef = {
+            name: fixture.name,
+            userToken: "context1",
+            expext: fixture.expected,
+            gradeNames: "gpii.test.contextIntegration.testCaseHolder",
+            config: {
+                configName: "linux-builtIn-config",
+                configPath: "tests/platform/linux/configs"
+            },
+            sequence: []
+        };
+        fluid.each(fixture.sequenceSegments, function (arr) {
```

Comment:
Well, this is interesting, isn't it.... I guess there's no reason we shouldn't support this in the core framework

Although in general we need a much better way of talking about "the calculus of building up test sequences" rather than just sticking them in an array or even an array of arrays....